crates/pop-cli/tests/contract.rs

```diff
@@ -7,14 +7,52 @@ use pop_contracts::{
 	contracts_node_generator, dry_run_gas_estimate_instantiate, instantiate_smart_contract,
 	run_contracts_node, set_up_deployment, Contract, UpOpts,
 };
-use std::{path::Path, process::Command as Cmd};
+use serde::{Deserialize, Serialize};
+use std::{path::Path, process::Command as Cmd, time::Duration};
 use strum::VariantArray;
+use subxt::{config::DefaultExtrinsicParamsBuilder as Params, tx::Payload, utils::to_hex};
+use subxt_signer::sr25519::dev;
+use tokio::time::sleep;
 use url::Url;
 
+// This struct implements the [`Payload`] trait and is used to submit
+// pre-encoded SCALE call data directly, without the dynamic construction of transactions.
+struct CallData(Vec<u8>);
+impl Payload for CallData {
```

Comment:
leaving a note: I copied this from the pop call parachain PR (IIRC). Once everything gets merged in we should hopefully be able to just import this.